------------------------------------------------------------------------
r14964 | kostya | 2008-08-25 08:28:52 +0200 (Mo, 25. Aug 2008) | 4 lines

fir_32bands_perfect[] contents were mixed, restore original order
Patch by Alexander Patrakov ($lastname <whirlpool> gmail.com)
Thread: dcadata.h: fir_32bands_perfect[] is wrong

------------------------------------------------------------------------
Index: libavcodec/dcadata.h
===================================================================
--- a/libavcodec/dcadata.h	(Revision 14963)
+++ b/libavcodec/dcadata.h	(Revision 14964)
@@ -6285,516 +6285,516 @@
 static const float fir_32bands_perfect[] =
 {
 +1.135985195E-010,
++7.018770981E-011,
+-1.608403011E-008,
+-5.083275667E-008,
+-1.543309907E-007,
+-3.961981463E-007,
+-7.342250683E-007,
+-3.970030775E-007,
+-4.741137047E-007,
 -6.022448247E-007,
+-6.628192182E-007,
+-6.982898526E-007,
+-7.020648809E-007,
+-6.767839409E-007,
+-6.262345096E-007,
+-5.564140224E-007,
++7.003467317E-007,
++8.419976893E-007,
 +9.742954035E-007,
-+7.018770981E-011,
--6.628192182E-007,
 +1.085227950E-006,
--1.608403011E-008,
--6.982898526E-007,
 +1.162929266E-006,
--5.083275667E-008,
--7.020648809E-007,
 +1.194632091E-006,
--1.543309907E-007,
--6.767839409E-007,
 +1.179182050E-006,
--3.961981463E-007,
--6.262345096E-007,
 +1.033426656E-006,
--7.342250683E-007,
--5.564140224E-007,
 +9.451737242E-007,
--3.970030775E-007,
-+7.003467317E-007,
 +1.975324267E-006,
--4.741137047E-007,
-+8.419976893E-007,
 +1.190443072E-006,
 +5.234479659E-007,
++2.014677420E-007,
++7.834767501E-008,
+-6.702406963E-010,
+-1.613285505E-009,
+-2.682709610E-009,
+-3.399493131E-009,
++1.314406006E-008,
++7.506701927E-009,
++2.788728892E-008,
++1.444918922E-007,
++3.132386439E-007,
++1.399798180E-006,
++2.032118118E-006,
++2.715013807E-006,
++3.453840463E-006,
++4.195037945E-006,
++4.896494374E-006,
++5.516381407E-006,
++6.015239251E-006,
++6.361419310E-006,
++8.006985809E-006,
++8.087732567E-006,
++7.941360309E-006,
++7.568834008E-006,
++6.986399967E-006,
++6.225028756E-006,
++5.315936960E-006,
++4.429412002E-006,
++3.332600045E-006,
++8.427224429E-007,
++4.341498823E-007,
++9.458596395E-008,
++2.975164826E-008,
 +6.402664354E-008,
+-3.246264413E-008,
+-3.809887872E-008,
++8.434094667E-008,
++6.437721822E-008,
++1.189317118E-006,
++2.497214155E-006,
++3.617151151E-006,
++3.157242645E-006,
++2.319611212E-006,
++7.869333785E-006,
++9.826449968E-006,
++1.177108606E-005,
++1.379448349E-005,
++1.571428584E-005,
++1.743183020E-005,
++1.884208177E-005,
++1.987093310E-005,
++2.042970118E-005,
+-3.144468428E-005,
+-3.334947178E-005,
+-3.460439257E-005,
+-3.515914432E-005,
+-3.495384954E-005,
+-3.397853652E-005,
+-3.225446198E-005,
+-2.978993689E-005,
+-2.677291741E-005,
+-1.806914770E-005,
+-1.776598037E-005,
+-1.661818715E-005,
+-1.207003334E-005,
+-6.993315310E-006,
+-5.633860383E-007,
+-9.984935332E-007,
 -1.470520488E-006,
-+2.014677420E-007,
--3.246264413E-008,
 -1.853591357E-006,
-+7.834767501E-008,
--3.809887872E-008,
 +7.198007665E-007,
-+8.434094667E-008,
 +3.086857760E-006,
--6.702406963E-010,
-+6.437721822E-008,
 +6.084746474E-006,
--1.613285505E-009,
-+1.189317118E-006,
 +9.561075785E-006,
--2.682709610E-009,
-+2.497214155E-006,
 +1.309637537E-005,
--3.399493131E-009,
-+3.617151151E-006,
 +2.263354872E-005,
-+1.314406006E-008,
-+3.157242645E-006,
 +2.847247197E-005,
-+7.506701927E-009,
-+2.319611212E-006,
 +3.415624451E-005,
-+2.788728892E-008,
-+7.869333785E-006,
 +3.946387005E-005,
-+1.444918922E-007,
-+9.826449968E-006,
 +4.425736552E-005,
-+3.132386439E-007,
-+1.177108606E-005,
 +4.839275425E-005,
-+1.399798180E-006,
-+1.379448349E-005,
 +5.176846025E-005,
-+2.032118118E-006,
-+1.571428584E-005,
 +5.429694284E-005,
-+2.715013807E-006,
-+1.743183020E-005,
 +5.595519906E-005,
-+3.453840463E-006,
-+1.884208177E-005,
 +4.916387297E-006,
-+4.195037945E-006,
-+1.987093310E-005,
 +9.299508747E-006,
-+4.896494374E-006,
-+2.042970118E-005,
 +1.356193479E-005,
-+5.516381407E-006,
--3.144468428E-005,
 +1.751866148E-005,
-+6.015239251E-006,
--3.334947178E-005,
 +2.093936746E-005,
-+6.361419310E-006,
--3.460439257E-005,
 +2.362549276E-005,
-+8.006985809E-006,
--3.515914432E-005,
 +2.537086584E-005,
-+8.087732567E-006,
--3.495384954E-005,
 +2.618136386E-005,
-+7.941360309E-006,
--3.397853652E-005,
 +2.554462844E-005,
-+7.568834008E-006,
--3.225446198E-005,
 +3.018750249E-005,
-+6.986399967E-006,
--2.978993689E-005,
 +2.570833203E-005,
-+6.225028756E-006,
--2.677291741E-005,
 +1.985177369E-005,
-+5.315936960E-006,
--1.806914770E-005,
 +1.191342653E-005,
-+4.429412002E-006,
--1.776598037E-005,
 +2.525620175E-006,
-+3.332600045E-006,
--1.661818715E-005,
 -1.521241393E-005,
-+8.427224429E-007,
--1.207003334E-005,
 -1.617751332E-005,
-+4.341498823E-007,
--6.993315310E-006,
 +1.992636317E-005,
-+9.458596395E-008,
--5.633860383E-007,
 +1.774702469E-005,
-+2.975164826E-008,
--9.984935332E-007,
 +4.624524081E-005,
 +5.610509834E-005,
++6.568001118E-005,
++7.513730816E-005,
++8.413690375E-005,
++8.757545584E-005,
++9.517164290E-005,
++1.020687996E-004,
++1.084438481E-004,
++1.140582463E-004,
++1.187910311E-004,
++1.224978914E-004,
++1.250260248E-004,
++1.262027217E-004,
++1.226499153E-004,
++1.213575742E-004,
++1.180980107E-004,
++1.126275165E-004,
++1.047207043E-004,
++9.417100227E-005,
++8.078388782E-005,
++6.447290798E-005,
++4.491530854E-005,
++2.470704203E-005,
+-1.714242217E-006,
+-3.193307566E-005,
+-6.541742187E-005,
+-1.024175072E-004,
+-1.312203676E-004,
+-1.774113771E-004,
+-2.233728592E-004,
+-2.682086197E-004,
+-3.347633174E-004,
+-3.906481725E-004,
+-4.490280990E-004,
+-5.099929986E-004,
 -5.729619297E-004,
+-6.358824321E-004,
+-7.021900383E-004,
+-7.698345580E-004,
+-8.385353722E-004,
+-9.078957955E-004,
+-9.775133803E-004,
+-1.046945457E-003,
+-1.115717343E-003,
+-1.183370827E-003,
+-1.252829796E-003,
+-1.316190348E-003,
+-1.376571832E-003,
+-1.433344092E-003,
+-1.485876855E-003,
+-1.533520175E-003,
+-1.575609902E-003,
+-1.611457788E-003,
+-1.640390139E-003,
+-1.661288203E-003,
+-1.674512983E-003,
+-1.678415807E-003,
+-1.672798418E-003,
+-1.656501088E-003,
+-1.633993932E-003,
+-1.593449386E-003,
++1.542080659E-003,
++1.479332102E-003,
++1.395521569E-003,
++1.303116791E-003,
++1.196175464E-003,
++1.073757303E-003,
++9.358961834E-004,
++7.817269652E-004,
++6.114174030E-004,
 +4.244441516E-004,
-+6.568001118E-005,
--6.358824321E-004,
 +2.206075296E-004,
-+7.513730816E-005,
--7.021900383E-004,
 -2.719412748E-007,
-+8.413690375E-005,
--7.698345580E-004,
 -2.382978710E-004,
-+8.757545584E-005,
--8.385353722E-004,
 -4.935106263E-004,
-+9.517164290E-005,
--9.078957955E-004,
 -7.658848190E-004,
-+1.020687996E-004,
--9.775133803E-004,
 -1.055365428E-003,
-+1.084438481E-004,
--1.046945457E-003,
 -1.361547387E-003,
-+1.140582463E-004,
--1.115717343E-003,
 -1.684492454E-003,
-+1.187910311E-004,
--1.183370827E-003,
 -2.023874084E-003,
-+1.224978914E-004,
--1.252829796E-003,
 -2.379294252E-003,
-+1.250260248E-004,
--1.316190348E-003,
 -2.750317100E-003,
-+1.262027217E-004,
--1.376571832E-003,
 -3.136433195E-003,
-+1.226499153E-004,
--1.433344092E-003,
 -3.537061159E-003,
-+1.213575742E-004,
--1.485876855E-003,
 -3.951539751E-003,
-+1.180980107E-004,
--1.533520175E-003,
 -4.379155114E-003,
-+1.126275165E-004,
--1.575609902E-003,
 -4.819062538E-003,
-+1.047207043E-004,
--1.611457788E-003,
 -5.270531867E-003,
-+9.417100227E-005,
--1.640390139E-003,
 -5.732392892E-003,
-+8.078388782E-005,
--1.661288203E-003,
 -6.203945260E-003,
-+6.447290798E-005,
--1.674512983E-003,
 -6.683901884E-003,
-+4.491530854E-005,
--1.678415807E-003,
 -7.170005701E-003,
-+2.470704203E-005,
--1.672798418E-003,
 -7.664063945E-003,
--1.714242217E-006,
--1.656501088E-003,
 -8.162760176E-003,
--3.193307566E-005,
--1.633993932E-003,
 -8.665001951E-003,
--6.541742187E-005,
--1.593449386E-003,
 -9.170533158E-003,
--1.024175072E-004,
-+1.542080659E-003,
 -9.676489048E-003,
--1.312203676E-004,
-+1.479332102E-003,
 -1.018219907E-002,
--1.774113771E-004,
-+1.395521569E-003,
 -1.068630442E-002,
--2.233728592E-004,
-+1.303116791E-003,
 -1.118756086E-002,
--2.682086197E-004,
-+1.196175464E-003,
 -1.168460958E-002,
--3.347633174E-004,
-+1.073757303E-003,
 -1.217562053E-002,
--3.906481725E-004,
-+9.358961834E-004,
 -1.265939046E-002,
--4.490280990E-004,
-+7.817269652E-004,
 -1.313448418E-002,
--5.099929986E-004,
-+6.114174030E-004,
 -1.359948888E-002,
 -1.405300573E-002,
+-1.449365262E-002,
+-1.492007636E-002,
+-1.533095632E-002,
+-1.572482102E-002,
+-1.610082202E-002,
+-1.645756140E-002,
+-1.679391414E-002,
+-1.710879989E-002,
+-1.740120351E-002,
+-1.767017506E-002,
+-1.791484281E-002,
+-1.813439466E-002,
+-1.832821220E-002,
+-1.849545911E-002,
+-1.863567345E-002,
+-1.874836907E-002,
+-1.883326657E-002,
+-1.889026538E-002,
+-1.891860925E-002,
++1.891860925E-002,
++1.889026538E-002,
++1.883326657E-002,
++1.874836907E-002,
++1.863567345E-002,
++1.849545911E-002,
++1.832821220E-002,
++1.813439466E-002,
++1.791484281E-002,
++1.767017506E-002,
++1.740120351E-002,
++1.710879989E-002,
++1.679391414E-002,
++1.645756140E-002,
++1.610082202E-002,
 +1.572482102E-002,
++1.533095632E-002,
++1.492007636E-002,
++1.449365262E-002,
++1.405300573E-002,
++1.359948888E-002,
++1.313448418E-002,
++1.265939046E-002,
++1.217562053E-002,
++1.168460958E-002,
++1.118756086E-002,
++1.068630442E-002,
++1.018219907E-002,
++9.676489048E-003,
++9.170533158E-003,
++8.665001951E-003,
++8.162760176E-003,
++7.664063945E-003,
++7.170005701E-003,
++6.683901884E-003,
++6.203945260E-003,
++5.732392892E-003,
++5.270531867E-003,
++4.819062538E-003,
++4.379155114E-003,
++3.951539751E-003,
++3.537061159E-003,
++3.136433195E-003,
++2.750317100E-003,
++2.379294252E-003,
++2.023874084E-003,
++1.684492454E-003,
++1.361547387E-003,
++1.055365428E-003,
++7.658848190E-004,
 +4.935106263E-004,
--1.449365262E-002,
-+1.533095632E-002,
 +2.382978710E-004,
--1.492007636E-002,
-+1.492007636E-002,
 +2.719412748E-007,
--1.533095632E-002,
-+1.449365262E-002,
 -2.206075296E-004,
--1.572482102E-002,
-+1.405300573E-002,
 -4.244441516E-004,
--1.610082202E-002,
-+1.359948888E-002,
 -6.114174030E-004,
--1.645756140E-002,
-+1.313448418E-002,
 -7.817269652E-004,
--1.679391414E-002,
-+1.265939046E-002,
 -9.358961834E-004,
--1.710879989E-002,
-+1.217562053E-002,
 -1.073757303E-003,
--1.740120351E-002,
-+1.168460958E-002,
 -1.196175464E-003,
--1.767017506E-002,
-+1.118756086E-002,
 -1.303116791E-003,
--1.791484281E-002,
-+1.068630442E-002,
 -1.395521569E-003,
--1.813439466E-002,
-+1.018219907E-002,
 -1.479332102E-003,
--1.832821220E-002,
-+9.676489048E-003,
 -1.542080659E-003,
--1.849545911E-002,
-+9.170533158E-003,
 +1.593449386E-003,
--1.863567345E-002,
-+8.665001951E-003,
 +1.633993932E-003,
--1.874836907E-002,
-+8.162760176E-003,
 +1.656501088E-003,
--1.883326657E-002,
-+7.664063945E-003,
 +1.672798418E-003,
--1.889026538E-002,
-+7.170005701E-003,
 +1.678415807E-003,
--1.891860925E-002,
-+6.683901884E-003,
 +1.674512983E-003,
-+1.891860925E-002,
-+6.203945260E-003,
 +1.661288203E-003,
-+1.889026538E-002,
-+5.732392892E-003,
 +1.640390139E-003,
-+1.883326657E-002,
-+5.270531867E-003,
 +1.611457788E-003,
-+1.874836907E-002,
-+4.819062538E-003,
 +1.575609902E-003,
-+1.863567345E-002,
-+4.379155114E-003,
 +1.533520175E-003,
-+1.849545911E-002,
-+3.951539751E-003,
 +1.485876855E-003,
-+1.832821220E-002,
-+3.537061159E-003,
 +1.433344092E-003,
-+1.813439466E-002,
-+3.136433195E-003,
 +1.376571832E-003,
-+1.791484281E-002,
-+2.750317100E-003,
 +1.316190348E-003,
-+1.767017506E-002,
-+2.379294252E-003,
 +1.252829796E-003,
-+1.740120351E-002,
-+2.023874084E-003,
 +1.183370827E-003,
-+1.710879989E-002,
-+1.684492454E-003,
 +1.115717343E-003,
-+1.679391414E-002,
-+1.361547387E-003,
 +1.046945457E-003,
-+1.645756140E-002,
-+1.055365428E-003,
 +9.775133803E-004,
-+1.610082202E-002,
-+7.658848190E-004,
 +9.078957955E-004,
 +8.385353722E-004,
++7.698345580E-004,
++7.021900383E-004,
++6.358824321E-004,
++5.729619297E-004,
++5.099929986E-004,
++4.490280990E-004,
++3.906481725E-004,
++3.347633174E-004,
++2.682086197E-004,
++2.233728592E-004,
++1.774113771E-004,
++1.312203676E-004,
++1.024175072E-004,
++6.541742187E-005,
++3.193307566E-005,
++1.714242217E-006,
+-2.470704203E-005,
+-4.491530854E-005,
+-6.447290798E-005,
+-8.078388782E-005,
+-9.417100227E-005,
+-1.047207043E-004,
+-1.126275165E-004,
+-1.180980107E-004,
+-1.213575742E-004,
+-1.226499153E-004,
+-1.262027217E-004,
+-1.250260248E-004,
+-1.224978914E-004,
+-1.187910311E-004,
+-1.140582463E-004,
+-1.084438481E-004,
+-1.020687996E-004,
+-9.517164290E-005,
 -8.757545584E-005,
+-8.413690375E-005,
+-7.513730816E-005,
+-6.568001118E-005,
+-5.610509834E-005,
+-4.624524081E-005,
+-1.774702469E-005,
+-1.992636317E-005,
++1.617751332E-005,
++1.521241393E-005,
+-2.525620175E-006,
+-1.191342653E-005,
+-1.985177369E-005,
+-2.570833203E-005,
+-3.018750249E-005,
+-2.554462844E-005,
+-2.618136386E-005,
+-2.537086584E-005,
+-2.362549276E-005,
+-2.093936746E-005,
+-1.751866148E-005,
+-1.356193479E-005,
+-9.299508747E-006,
+-4.916387297E-006,
+-5.595519906E-005,
+-5.429694284E-005,
+-5.176846025E-005,
+-4.839275425E-005,
+-4.425736552E-005,
+-3.946387005E-005,
+-3.415624451E-005,
+-2.847247197E-005,
+-2.263354872E-005,
+-1.309637537E-005,
+-9.561075785E-006,
 -6.084746474E-006,
-+7.698345580E-004,
--8.413690375E-005,
 -3.086857760E-006,
-+7.021900383E-004,
--7.513730816E-005,
 -7.198007665E-007,
-+6.358824321E-004,
--6.568001118E-005,
 +1.853591357E-006,
-+5.729619297E-004,
--5.610509834E-005,
 +1.470520488E-006,
-+5.099929986E-004,
--4.624524081E-005,
 +9.984935332E-007,
-+4.490280990E-004,
--1.774702469E-005,
 +5.633860383E-007,
-+3.906481725E-004,
--1.992636317E-005,
 +6.993315310E-006,
-+3.347633174E-004,
-+1.617751332E-005,
 +1.207003334E-005,
-+2.682086197E-004,
-+1.521241393E-005,
 +1.661818715E-005,
-+2.233728592E-004,
--2.525620175E-006,
 +1.776598037E-005,
-+1.774113771E-004,
--1.191342653E-005,
 +1.806914770E-005,
-+1.312203676E-004,
--1.985177369E-005,
 +2.677291741E-005,
-+1.024175072E-004,
--2.570833203E-005,
 +2.978993689E-005,
-+6.541742187E-005,
--3.018750249E-005,
 +3.225446198E-005,
-+3.193307566E-005,
--2.554462844E-005,
 +3.397853652E-005,
-+1.714242217E-006,
--2.618136386E-005,
 +3.495384954E-005,
--2.470704203E-005,
--2.537086584E-005,
 +3.515914432E-005,
--4.491530854E-005,
--2.362549276E-005,
 +3.460439257E-005,
--6.447290798E-005,
--2.093936746E-005,
 +3.334947178E-005,
--8.078388782E-005,
--1.751866148E-005,
 +3.144468428E-005,
--9.417100227E-005,
--1.356193479E-005,
 -2.042970118E-005,
--1.047207043E-004,
--9.299508747E-006,
 -1.987093310E-005,
--1.126275165E-004,
--4.916387297E-006,
 -1.884208177E-005,
--1.180980107E-004,
--5.595519906E-005,
 -1.743183020E-005,
--1.213575742E-004,
--5.429694284E-005,
 -1.571428584E-005,
--1.226499153E-004,
--5.176846025E-005,
 -1.379448349E-005,
--1.262027217E-004,
--4.839275425E-005,
 -1.177108606E-005,
--1.250260248E-004,
--4.425736552E-005,
 -9.826449968E-006,
--1.224978914E-004,
--3.946387005E-005,
 -7.869333785E-006,
--1.187910311E-004,
--3.415624451E-005,
 -2.319611212E-006,
--1.140582463E-004,
--2.847247197E-005,
 -3.157242645E-006,
--1.084438481E-004,
--2.263354872E-005,
 -3.617151151E-006,
--1.020687996E-004,
--1.309637537E-005,
 -2.497214155E-006,
--9.517164290E-005,
--9.561075785E-006,
 -1.189317118E-006,
 -6.437721822E-008,
+-8.434094667E-008,
++3.809887872E-008,
++3.246264413E-008,
+-6.402664354E-008,
+-2.975164826E-008,
+-9.458596395E-008,
+-4.341498823E-007,
+-8.427224429E-007,
+-3.332600045E-006,
+-4.429412002E-006,
+-5.315936960E-006,
+-6.225028756E-006,
+-6.986399967E-006,
+-7.568834008E-006,
+-7.941360309E-006,
+-8.087732567E-006,
+-8.006985809E-006,
+-6.361419310E-006,
+-6.015239251E-006,
+-5.516381407E-006,
+-4.896494374E-006,
 -4.195037945E-006,
+-3.453840463E-006,
+-2.715013807E-006,
+-2.032118118E-006,
+-1.399798180E-006,
+-3.132386439E-007,
+-1.444918922E-007,
+-2.788728892E-008,
+-7.506701927E-009,
+-1.314406006E-008,
++3.399493131E-009,
++2.682709610E-009,
++1.613285505E-009,
++6.702406963E-010,
+-7.834767501E-008,
+-2.014677420E-007,
+-5.234479659E-007,
+-1.190443072E-006,
+-1.975324267E-006,
+-9.451737242E-007,
+-1.033426656E-006,
+-1.179182050E-006,
 -1.194632091E-006,
--8.434094667E-008,
--3.453840463E-006,
 -1.162929266E-006,
-+3.809887872E-008,
--2.715013807E-006,
 -1.085227950E-006,
-+3.246264413E-008,
--2.032118118E-006,
 -9.742954035E-007,
--6.402664354E-008,
--1.399798180E-006,
 -8.419976893E-007,
--2.975164826E-008,
--3.132386439E-007,
 -7.003467317E-007,
--9.458596395E-008,
--1.444918922E-007,
 +5.564140224E-007,
--4.341498823E-007,
--2.788728892E-008,
 +6.262345096E-007,
--8.427224429E-007,
--7.506701927E-009,
 +6.767839409E-007,
--3.332600045E-006,
--1.314406006E-008,
 +7.020648809E-007,
--4.429412002E-006,
-+3.399493131E-009,
 +6.982898526E-007,
--5.315936960E-006,
-+2.682709610E-009,
 +6.628192182E-007,
--6.225028756E-006,
-+1.613285505E-009,
 +6.022448247E-007,
--6.986399967E-006,
-+6.702406963E-010,
 +4.741137047E-007,
--7.568834008E-006,
--7.834767501E-008,
 +3.970030775E-007,
--7.941360309E-006,
--2.014677420E-007,
 +7.342250683E-007,
--8.087732567E-006,
--5.234479659E-007,
 +3.961981463E-007,
--8.006985809E-006,
--1.190443072E-006,
 +1.543309907E-007,
--6.361419310E-006,
--1.975324267E-006,
 +5.083275667E-008,
--6.015239251E-006,
--9.451737242E-007,
 +1.608403011E-008,
--5.516381407E-006,
--1.033426656E-006,
 -7.018770981E-011,
--4.896494374E-006,
--1.179182050E-006,
 -1.135985195E-010
 };
 
